onnxruntime/core/providers/nnapi/nnapi_builtin/builders/shaper.cc

```diff
@@ -388,6 +386,31 @@ Status Shaper::ConcatImpl(const std::vector<std::string>& input_names,
   return Status::OK();
 }
 
+Status Shaper::SplitImpl(const std::string& input_name, int32_t axis, int32_t count,
+                         const std::vector<std::string>& output_names) {
+  ORT_RETURN_IF_NOT(count > 0, "count [", count, "] must be greater than 0");
+  ORT_RETURN_IF_NOT(static_cast<size_t>(count) == output_names.size(),
+                    "count [", count, "] must match the number of outputs [", output_names.size(), "]");
+
+  const auto& input_shape = shape_map_.at(input_name);
+  const auto& input_rank = input_shape.size();
```

Comment:
nit: maybe irrelevant if Shaper can be removed, but what does `at` do if the key isn't found? In some places like std::vector it terminates with no hint as to why. If it does the same here, and we want/need to check the key validity, we should do it ourselves so we get error output.

Reply:
I believe it will terminate if exceptions are disabled. perhaps we can make some at()-like wrapper that provides some output. given the amount of map::at() usage here, maybe we can address this separately or just remove Shaper